### PR TITLE
fixes for latest codespell release

### LINKED ISSUE #11045: bunch of codespell errors (and false alarms)
I suspect maybe the codespell version got updated: CIs just started (?) failing like this:

https://github.com/mne-tools/mne-python/runs/7889067340?check_suite_focus=true#step:8:31

```
Error: mne/evoked.py:275: basline ==> baseline
Error: mne/conftest.py:361: nAm ==> name
Error: mne/conftest.py:361: nAm ==> name
Error: mne/dipole.py:543: nAm ==> name
Error: mne/dipole.py:607: nAm ==> name
Error: mne/dipole.py:607: nAm ==> name
Error: mne/dipole.py:607: nAm ==> name
Error: mne/dipole.py:607: nAm ==> name
Error: mne/coreg.py:1767: enfore ==> enforce
Error: mne/simulation/source.py:107: nAm ==> name
Error: mne/preprocessing/tests/test_eeglab_infomax.py:79: comparasion ==> comparison
Error: mne/forward/tests/test_make_forward.py:427: nAm ==> name
Error: mne/decoding/ssd.py:187: ording ==> ordering
Error: mne/io/brainvision/brainvision.py:934: fom ==> from
Error: mne/io/brainvision/brainvision.py:996: fom ==> from
Error: mne/io/curry/curry.py:396: enttry ==> entry
Error: mne/io/nihon/nihon.py:395: acqusition ==> acquisition
Error: mne/utils/check.py:813: nAm ==> name
Error: mne/utils/check.py:816: nAm ==> name
Error: mne/viz/misc.py:808: nAm ==> name
Error: mne/viz/_3d.py:2866: nAm ==> name
Error: doc/conf.py:726: bu ==> by
Error: doc/_includes/institutional-partners.rst:33: bu ==> by
Error: doc/changes/0.20.inc:162: nAm ==> name
Error: doc/changes/0.14.inc:172: nAm ==> name
Error: tutorials/simulation/80_dics.py:87: nAm ==> name
Error: tutorials/inverse/80_brainstorm_phantom_elekta.py:35: nAm ==> name
Error: tutorials/inverse/80_brainstorm_phantom_elekta.py:137: nAm ==> name
Error: tutorials/inverse/80_brainstorm_phantom_elekta.py:157: nAm ==> name
Error: tutorials/stats-sensor-space/75_cluster_ftest_spatiotemporal.py:144: possbile ==> possible, possibly
Error: examples/simulation/simulate_raw_data.py:70: nAm ==> name
Error: examples/simulation/simulated_raw_data_using_subject_anatomy.py:97: nAm ==> name
Error: examples/simulation/source_simulator.py:63: nAm ==> name
Error: examples/inverse/mixed_source_space_inverse.py:178: nAm ==> name
```

most of these will go away if we special-case nAm (nanoAmperes) but some of them appear legit.

### PR DESCRIPTION
**Pull request by drammock**
closes #11045

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@
 PYTHON ?= python
 PYTESTS ?= py.test
 CTAGS ?= ctags
-CODESPELL_SKIPS ?= "doc/auto_*,*.fif,*.eve,*.gz,*.tgz,*.zip,*.mat,*.stc,*.label,*.w,*.bz2,*.annot,*.sulc,*.log,*.local-copy,*.orig_avg,*.inflated_avg,*.gii,*.pyc,*.doctree,*.pickle,*.inv,*.png,*.edf,*.touch,*.thickness,*.nofix,*.volume,*.defect_borders,*.mgh,lh.*,rh.*,COR-*,FreeSurferColorLUT.txt,*.examples,.xdebug_mris_calc,bad.segments,BadChannels,*.hist,empty_file,*.orig,*.js,*.map,*.ipynb,searchindex.dat,install_mne_c.rst,plot_*.rst,*.rst.txt,c_EULA.rst*,*.html,gdf_encodes.txt,*.svg,references.bib,*.css"
+CODESPELL_SKIPS ?= "doc/_build,doc/auto_*,*.fif,*.eve,*.gz,*.tgz,*.zip,*.mat,*.stc,*.label,*.w,*.bz2,*.annot,*.sulc,*.log,*.local-copy,*.orig_avg,*.inflated_avg,*.gii,*.pyc,*.doctree,*.pickle,*.inv,*.png,*.edf,*.touch,*.thickness,*.nofix,*.volume,*.defect_borders,*.mgh,lh.*,rh.*,COR-*,FreeSurferColorLUT.txt,*.examples,.xdebug_mris_calc,bad.segments,BadChannels,*.hist,empty_file,*.orig,*.js,*.map,*.ipynb,searchindex.dat,install_mne_c.rst,plot_*.rst,*.rst.txt,c_EULA.rst*,*.html,gdf_encodes.txt,*.svg,references.bib,*.css"
```
Comment (drammock) [inline Makefile:6]: ignore files in `doc/_build`

File: Makefile [hunk 2]
```diff
@@ -103,10 +103,10 @@ flake:
 	@echo "flake8 passed"
 
 codespell:  # running manually
-	@codespell --builtin clear,rare,informal,names,usage -w -i 3 -q 3 -S $(CODESPELL_SKIPS) --ignore-words=ignore_words.txt $(CODESPELL_DIRS)
+	@codespell --builtin clear,rare,informal,names,usage -w -i 3 -q 3 -S $(CODESPELL_SKIPS) --ignore-words=ignore_words.txt --uri-ignore-words-list=bu $(CODESPELL_DIRS)
```
Comment (drammock) [inline Makefile:106]: we have a couple of URLs for Boston University, codespell thinks `bu` -> `by`

File: ignore_words.txt [hunk 3]
```diff
@@ -37,3 +37,4 @@ nin
 trough
 recuse
 ro
+nam
```
Comment (drammock) [inline ignore_words.txt:40]: this was the worst one, we have lots of nanoAmperes (nAm) in our plot titles / code comments.

**Review by drammock — COMMENTED** [review]
I've highlighted the config changes, everything else is just the actual spelling corrections

File: Makefile (hunk 2 above)
Comment (drammock) [reply Makefile:106]: figured it out. Hopefully this should fix it: https://github.com/GuillaumeFavelier/actions-codespell/pull/2

ping @GuillaumeFavelier

**Comment by larsoner** [comment]
To make it easier we could just add `bu` to the ignore list. Looks like it's not working properly in the URI list for some reason

**Comment by larsoner** [comment]
... if you feel like fixing this here, too:
```
mne/utils/dataframe.py:69: in _build_data_frame
    df.set_index(index, inplace=True)
...
E   FutureWarning: The 'inplace' keyword in DataFrame.set_index is deprecated and will be removed in a future version. Use `df = df.set_index(..., copy=False)` instead.
```
feel free. Otherwise I can do it in another PR

**Comment by larsoner** [comment]
I'll push a quick commit for `pandas` and `bu` (the latter we can revert once the action is updated)

File: mne/utils/dataframe.py [hunk 4]
```diff
@@ -62,17 +63,21 @@ def _build_data_frame(inst, data, picks, long_format, mindex, index,
     for i, (k, v) in enumerate(mindex):
         df.insert(i, k, v)
     # build Index
+    kwargs = dict()
+    key = 'copy' if 'copy' in _get_args(df.set_index) else 'inplace'
+    val = False if key == 'copy' else True
+    kwargs[key] = val
     if long_format:
-        df.set_index(default_index, inplace=True)
+        df = df.set_index(default_index, **kwargs)
```
Comment (drammock) [inline mne/utils/dataframe.py:71]: this won't work because `df.set_index` will return `None` when `inplace=True`.  I'll fix it.

Comment (larsoner) [reply mne/utils/dataframe.py]: Yeah, this is true on old Pandas at lea,st. When `copy=False` it actually returns an instance now, and you *must* use it (the original is not in the correct state anymore). I pushed a commit to fix it already

File: mne/utils/dataframe.py [hunk 5]
```diff
@@ -47,6 +48,16 @@ def _convert_times(inst, times, time_format):
...
 
 
+def _inplace(df, method, **kwargs):
+    """Handle transition: inplace=True (pandas <1.5) → copy=False (>=1.5)."""
+    _meth = getattr(df, method)  # used for set_index() and rename()
+    if 'copy' in _get_args(_meth):
+        return _meth(**kwargs, copy=False)
+    else:
+        _meth(**kwargs, inplace=True)
+        return df
```
[the remark this answers is not in the document]
Comment (larsoner) [reply mne/utils/dataframe.py:58]: Yeah it's nicer than what I did

**Pull request merged**